cmd/antrea-agent/agent.go

```diff
@@ -126,6 +126,7 @@ func run(o *Options) error {
 	agentMonitor := monitor.NewAgentMonitor(crdClient, o.config.OVSBridge, nodeConfig.Name, nodeConfig.PodCIDR.String(), ifaceStore, ofClient, ovsBridgeClient)
 
 	go agentMonitor.Run(stopCh)
+	defer agentMonitor.DeleteAgentCRD(nodeConfig.Name)
```

Comment:
I would not delete the CRDs when Agent/Controller stops, otherwise UI could not show any info for temporary crashes/failures. Instead, I think Controller should remove Agent CRDs after a Node is removed from the Cluster.
And for the uninstallation case, we should have a separate mechanism to clean up all resources on the Node and K8s API, including CRDs.

Reply:
OK, got it this time. I think I have misunderstood the issue and thought we need to delete CRD whenever the Pod is deleted. If it is only for the case of a node leaving, then Quan's suggestion looks good to me. Will investigate and add the logic from controller side.